tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts

```diff
@@ -1285,20 +1308,245 @@ describe('scalar card', () => {
+            }
+          ),
+          buildTooltipDatum(
+            {
+              id: 'row2',
+              type: SeriesType.ORIGINAL,
+              displayName: 'Row 2',
+              visible: true,
+              color: '#0f0',
+              aux: false,
+            },
+            {
+              x: 1000,
+              step: 1000,
+              y: -500,
+              value: -500,
+              wallTime: new Date('2020-12-31').getTime(),
+            }
+          ),
+          buildTooltipDatum(
+            {
+              id: 'row3',
+              type: SeriesType.ORIGINAL,
+              displayName: 'Row 3',
+              visible: true,
+              color: '#00f',
+              aux: false,
+            },
+            {
+              x: 10000,
+              step: 10000,
+              y: 3,
+              value: 3,
+              wallTime: new Date('2021-01-01').getTime(),
+            }
+          ),
+        ]);
+        fixture.detectChanges();
+
+        assertTooltipRows(fixture, [
+          ['', 'Row 2', '-500', '1,000', jasmine.any(String)],
+          ['', 'Row 3', '3', '10,000', jasmine.any(String)],
+          ['', 'Row 1', '1000', '10', jasmine.any(String)],
+        ]);
+      }));
+
+      it('sorts by descending', fakeAsync(() => {
+        store.overrideSelector(
+          selectors.getMetricsTooltipSort,
+          TooltipSort.DESCENDING
+        );
+        store.overrideSelector(selectors.getMetricsScalarSmoothing, 0);
+        const fixture = createComponent('card1');
+        setTooltipData(fixture, [
+          buildTooltipDatum(
+            {
+              id: 'row1',
+              type: SeriesType.ORIGINAL,
+              displayName: 'Row 1',
+              visible: true,
+              color: '#f00',
+              aux: false,
+            },
+            {
+              x: 10,
+              step: 10,
+              y: 1000,
+              value: 1000,
+              wallTime: new Date('2020-01-01').getTime(),
+            }
+          ),
+          buildTooltipDatum(
+            {
+              id: 'row2',
+              type: SeriesType.ORIGINAL,
+              displayName: 'Row 2',
+              visible: true,
+              color: '#0f0',
+              aux: false,
+            },
+            {
+              x: 1000,
+              step: 1000,
+              y: -500,
+              value: -500,
+              wallTime: new Date('2020-12-31').getTime(),
+            }
+          ),
+          buildTooltipDatum(
+            {
+              id: 'row3',
+              type: SeriesType.ORIGINAL,
+              displayName: 'Row 3',
+              visible: true,
+              color: '#00f',
+              aux: false,
+            },
+            {
+              x: 10000,
+              step: 10000,
+              y: 3,
+              value: 3,
+              wallTime: new Date('2021-01-01').getTime(),
+            }
+          ),
```

Comment:
At some point we may want a set of test helpers to build these without
duplicating the same 51 lines in each function… but no action required
for this PR.